### PR TITLE
Enforce plc stage conventions

### PR DESCRIPTION
**Pull request by mehalshah**
For PLC scripts, either all stages should have flex categories or none of them should. Additionally, flex categories should be grouped together.

### CODE REVIEW

File: dashboard/app/models/script.rb [hunk 1]
```diff
@@ -63,6 +63,17 @@ class Script < ActiveRecord::Base
 
   def generate_plc_objects
     if professional_learning_course?
+      # Verify that either no stages have flex categories or they all have categories we recognize
+      learning_module_types = stages.map(&:flex_category)
+
+      unless learning_module_types.uniq == [nil] || Set.new(learning_module_types) <= Set.new(Plc::LearningModule::MODULE_TYPES)
+        raise "#{name}, either all stages must have flex categories that correspond to learning module types, or none of them should have categories"
```
Comment (Hamms) [inline dashboard/app/models/script.rb:70]: any reason why we're not raising a more-specific `RecordInvalid` error here and below?

File: dashboard/app/models/script.rb [hunk 2]
```diff
@@ -63,6 +63,17 @@ class Script < ActiveRecord::Base
...
+      end
+
+      unless learning_module_types.compact.empty? || learning_module_types.chunk { |x| x }.map(&:first) == learning_module_types.uniq
```
Comment (Hamms) [inline dashboard/app/models/script.rb:73]: isn't the `learning_module_types.compact.empty?` case redundant here?

File: dashboard/test/models/script_test.rb [hunk 3]
```diff
@@ -483,7 +483,26 @@ def populate_cache_and_disconnect_db
...
+  end
+
+  test 'Cannot setup script where some of the stages are missing flex categories' do
+    script_file = File.join(self.class.fixture_path, 'test-bad-plc-script-missing-categories.script')
+    assert_raises RuntimeError do
+      Script.setup([script_file])
+    end
+  end
+
+  test 'Cannot setup script where some of the stages are missing flex categories' do
```
Comment (Hamms) [inline dashboard/test/models/script_test.rb]: looks like ya forgot to update the name of this test

File: dashboard/app/models/script.rb (hunk 1 above)
Comment (mehalshah) [reply dashboard/app/models/script.rb:70]: RecordInvalid happens when trying to persist an object to the DB. That's not happening here, so it's not really appropriate to raise that exception

File: dashboard/app/models/script.rb (hunk 2 above)
Comment (mehalshah) [reply dashboard/app/models/script.rb:73]: No - if the pd course has no flex categories, then learning_modules_types is [nil, nil, nil] in which case this is needed

File: dashboard/test/models/script_test.rb (hunk 3 above)
Comment (mehalshah) [reply dashboard/test/models/script_test.rb]: Will change

File: dashboard/app/models/script.rb (hunk 1 above)
Comment (mehalshah) [reply dashboard/app/models/script.rb:70]: Actually I might be able to just raise this

File: dashboard/app/models/script.rb (hunk 2 above)
Comment (Hamms) [reply dashboard/app/models/script.rb:73]: ah, I was falsely assuming that `learning_module_types.chunk { |x| x }.map(&:first)` would give us `[nil]` in that situation

**Pull request merged**